backend/src/auth/token.guard.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { TokenType } from './interface';
+import { Metadata } from './metadata';
+import { AppLogger } from '../logger/logger.service';
+
+/**
+ * Guard to allow access to endpoints annotated with the bcws token decorator
+ * Used for sending the approved list of ids to ediaries/bcws
+ */
+@Injectable()
+export class TokenGuard implements CanActivate {
+  constructor(
+    private reflector: Reflector,
+    private logger: AppLogger,
+  ) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const PublicEndpoint = this.reflector.getAllAndOverride<boolean>(
+      Metadata.PUBLIC_ENDPOINT,
+      [context.getHandler(), context.getClass()],
+    );
+
+    if (PublicEndpoint) {
+      return true;
+    }
+
+    const requiredTokens = this.reflector.get<TokenType[]>(
+      Metadata.TOKEN_TYPE,
+      context.getHandler(),
+    );
+
+    if (!requiredTokens) {
+      return true;
+    }
+
+    const request = context.switchToHttp().getRequest();
+    const authHeader = request.headers.authentication;
```

Comment:
```suggestion
    const authHeader = request.headers.authorization;
```

Reply:
I did this as "authorization" b/c the CHEFS docs used "authorization" for their key. But, I think we can rename the key, I had just followed the example from their documentation.